wa/framework/workload.py

```diff
@@ -214,50 +255,60 @@ def __init__(self, workload, target, setup_timeout=5 * 60, run_timeout=10 * 60):
         self.target = target
         self.setup_timeout = setup_timeout
         self.run_timeout = run_timeout
+        self.revent_recorder = ReventRecorder(self.target)
         self.on_target_revent_binary = self.target.get_workpath('revent')
-        self.on_target_setup_revent = self.target.get_workpath('{}.setup.revent'.format(self.target.name))
-        self.on_target_run_revent = self.target.get_workpath('{}.run.revent'.format(self.target.name))
+        self.on_target_setup_revent = self.target.get_workpath('{}.setup.revent'.format(self.target.model))
+        self.on_target_run_revent = self.target.get_workpath('{}.run.revent'.format(self.target.model))
         self.logger = logging.getLogger('revent')
         self.revent_setup_file = None
         self.revent_run_file = None
 
-    def init_resources(self, context):
-        self.revent_setup_file = context.resolver.get(ReventFile(self.workload, 'setup'))
-        self.revent_run_file = context.resolver.get(ReventFile(self.workload, 'run'))
-
-    def setup(self, context):
-        self._check_revent_files(context)
-        self.target.killall('revent')
-        command = '{} replay {}'.format(self.on_target_revent_binary, self.on_target_setup_revent)
-        self.target.execute(command, timeout=self.setup_timeout)
-
-    def run(self, context):
-        command = '{} replay {}'.format(self.on_target_revent_binary, self.on_target_run_revent)
-        self.logger.debug('Replaying {}'.format(os.path.basename(self.on_target_run_revent)))
-        self.target.execute(command, timeout=self.run_timeout)
+    def init_resources(self, resolver):
+        self.revent_setup_file = resolver.get(ReventFile(owner=self.workload,
+                                                         stage='setup',
+                                                         target=self.target.model))
+        self.revent_run_file = resolver.get(ReventFile(owner=self.workload,
+                                                       stage='run',
+                                                       target=self.target.model))
```

Comment:
The `setup` recording should be optional. There should also be optional recording for `extract_results` and `teardown` stages.